controllers/helpers.go

```diff
@@ -1023,3 +1026,33 @@ func MachinePoolToAzureManagedControlPlaneMapFunc(ctx context.Context, c client.
 		return nil
 	}
 }
+
+// ClusterUpdatePauseChange returns a predicate that returns true for an update event when a cluster's
+// Spec.Paused changes between any two distinct values.
+func ClusterUpdatePauseChange(logger logr.Logger) predicate.Funcs {
```

Comment:
This is mostly copy-paste from CAPI's [`ClusterUpdateUnpaused`](https://github.com/kubernetes-sigs/cluster-api/blob/10d44b6ac7603f4aafd72dced33ae2c6915b7f04/util/predicates/cluster_predicates.go#L123).